EIPS/eip-7704.md

```diff
@@ -0,0 +1,137 @@
+---
+eip: 7704
+title: Adjust the BLOCKHASH opcode to read from storage and update its gas cost
+description: Serve the `BLOCKHASH (0x40)` opcode from the system contract storage from EIP-2935 to allow for stateless execution. Also adjust its gas cost to reflect storage access.
+author: Vitalik Buterin (@vbuterin), Tomasz Stanczak (@tkstanczak), Guillaume Ballet (@gballet), Gajinder Singh (@g11tech), Tanishq Jasoria (@tanishqjasoria), Ignacio Hagopian (@jsign), Jochem Brouwer (@jochem-brouwer), Gabriel Rocheleau (@gabrocheleau)
+discussions-to: https://ethereum-magicians.org/t/eip-2935-save-historical-block-hashes-in-state/4565
```

Comment:
Please create a discussions topic on Eth Magicians for this EIP

Reply:
Created the discussion here: https://ethereum-magicians.org/t/eip-7709-read-blockhash-opcode-from-storage-and-adjust-gas-cost/20052

Have also updated the link in the EIP.